modules/local/bigstitcher/module/main.nf

```diff
@@ -0,0 +1,71 @@
+process BIGSTITCHER_MODULE {
+    tag { meta.id }
+    container { task.ext.container ?: 'ghcr.io/janeliascicomp/bigstitcher:2.4.1-spark3.2.1-jdk8' }
+    cpus { spark.driver_cores }
+    memory { spark.driver_memory }
+
+    input:
+    tuple val(meta), path(bigstitcher_container), val(spark)
+    val(bigstitcher_container_val)
+    val(module_class)
+    val(module_args)
+
+    output:
+    tuple val(meta), env(full_bigstitcher_container), val(spark)
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def extra_args = module_args ?: ''
+    def executor_memory = spark.executor_memory.replace(" KB",'k').replace(" MB",'m').replace(" GB",'g').replace(" TB",'t')
+    def driver_memory = spark.driver_memory.replace(" KB",'k').replace(" MB",'m').replace(" GB",'g').replace(" TB",'t')
+    def app_jar = '/app/app.jar'
+    def full_bigstitcher_container_uri = get_input_uri(bigstitcher_container_val)
+    """
+    # if the fusion container is a Google bucket, S3 bucket, or an HTTP URI, use it as is
```

Comment:
Is this only for fusion? It looks like a generic way to call modules...

Reply:
Now it really is more generic and can handle more modules